packages/calypso-products/src/is-big-sky-onboarding.ts

```diff
@@ -0,0 +1,26 @@
+import { getPlansListExperiment } from './experiments';
+
+declare global {
+	interface Window {
+		__a8cBigSkyOnboarding?: boolean;
+	}
+}
+
+export function isBigSkyOnboarding(): boolean {
+	if ( new URLSearchParams( location.search ).has( 'big-sky-checkout' ) ) {
+		return true;
+	}
+
+	if (
+		getPlansListExperiment( 'calypso_signup_onboarding_goals_first_bigsky_202501_v1' ) !==
+		'treatment'
+	) {
+		return false;
+	}
+
+	if ( typeof window === 'undefined' ) {
+		return false;
+	}
+
+	return Boolean( window.__a8cBigSkyOnboarding );
```

Comment:
@vykes-mac  I discovered that the Global Styles code was doing a hack like this to conditionally show the text
https://github.com/Automattic/wp-calypso/blob/246305e458b570fca052f34579cc3ea0baee283a/packages/calypso-products/src/is-global-styles-on-personal-enabled.ts#L6

So I've done something similar 🙈

Reply:
I did this approach also, similar to global styles, but I opted for the experiment name approach so I didn't have to pass a parameter to the checkout page, but that works as well.